sklearn/neighbors/_base.py

```diff
@@ -1278,6 +1283,53 @@ class from an array representing our data set and ask who's
                 delayed_query(X[s], radius, return_distance, sort_results=sort_results)
                 for s in gen_even_slices(X.shape[0], n_jobs)
             )
+            if return_distance:
+                neigh_ind, neigh_dist = tuple(zip(*chunked_results))
+                results = np.hstack(neigh_dist), np.hstack(neigh_ind)
+            else:
+                results = np.hstack(chunked_results)
+        elif self._fit_method == "kd_tree":
+            if issparse(X):
+                raise ValueError(
+                    "%s does not work with sparse matrices. Densify the data, "
+                    "or set algorithm='brute'" % self._fit_method
+                )
+
+            n_jobs = effective_n_jobs(self.n_jobs)
+            delayed_query = delayed(self._sp_tree.query_ball_point)
+            chunked_results = Parallel(n_jobs, prefer="threads")(
+                delayed_query(X[s], radius, return_sorted=sort_results)
+                for s in gen_even_slices(X.shape[0], n_jobs)
+            )
+            nn_vals = []
+            for sub_arr in chunked_results[0]:
+                nn_vals.append(len(sub_arr))
+            if return_distance:
+                dd, ii = self._sp_tree.query(X, k=max(nn_vals), distance_upper_bound=radius)
+                dd_new = []
+                ii_new = []
+                for i in range(len(dd)):
+                    finite_indices = ii[i][np.isfinite(dd[i])]
+                    finite_dists = dd[i][np.isfinite(dd[i])]
+                    if sort_results:
+                        sort_inds = np.argsort(finite_indices)
+                        sorted_inds = finite_indices[sort_inds]
+                        sorted_dists = finite_dists[sort_inds]
```

Comment:
Different handling of ragged data structures and `inf`/invalid values also seems to be present, requiring these additional shims.